main.py

```diff
@@ -202,9 +206,19 @@ def ui_access_wrapper(self, *args, **kwargs):
         r'%s' % feconf.CREATOR_DASHBOARD_DATA_URL,
         creator_dashboard.CreatorDashboardHandler),
     get_redirect_route(
-        r'%s' % feconf.NEW_EXPLORATION_URL, creator_dashboard.NewExploration),
+        r'%s' % feconf.NEW_EXPLORATION_URL,
+        creator_dashboard.NewExplorationHandler),
     get_redirect_route(
-        r'%s' % feconf.NEW_COLLECTION_URL, creator_dashboard.NewCollection),
+        r'%s' % feconf.NEW_COLLECTION_URL,
+        creator_dashboard.NewCollectionHandler),
+    get_redirect_route(
```

Comment:
This is optional -- but if you want to be really sure that nothing comes in through these handlers, you can put these new routes in a separate list, and at the bottom, append that list to URLS_TO_SERVE if feconf.ENABLE_NEW_STRUCTURES is True. That will avoid you needing to put an individual check in each of the URLs, which could be error-prone if you miss one.

Just an FYI, it's up to you!

Reply:
I tried this way by having a URLS_TO_SERVE.extend([]) at the end, but a 404 is shown. Maybe the URLS_TO_SERVE are created before tests start and swap() doesn't affect it?